### PR TITLE
Add device detection while the tool is running

### PR DESCRIPTION
**Pull request by christianpape**
The tool will no allow to plug in and out the blink1 device 
after the tool was started. The device will be initialized
when found.

Please review @indigo423

### CODE REVIEW

File: src/main/java/org/opennms/onmsblink/OnmsBlink.java [hunk 1]
```diff
@@ -92,12 +85,15 @@ public boolean isConstantGlow() {
...
         }
-
+        
         @Override
         public void run() {
             while (true) {
                 if (maxSeverity.getId() <= 3 || isConstantGlow()) {
-                    blink1.fadeToRGB(0, colorMap.get(maxSeverity));
+                    System.out.println(blink1.fadeToRGB(0, colorMap.get(maxSeverity)));
+                    if (blink1.fadeToRGB(0, colorMap.get(maxSeverity)) == -1) {
```
Comment (indigo423) [inline src/main/java/org/opennms/onmsblink/OnmsBlink.java:94]: Deal with constant -1, 0, 3 correctly.

File: src/main/java/org/opennms/onmsblink/OnmsBlink.java [hunk 2]
```diff
@@ -107,7 +103,9 @@ public void run() {
                 } else {
                     int speed = 7 - Math.max(maxSeverity.getId(), 3);
 
-                    blink1.fadeToRGB(100 + speed * 100, colorMap.get(maxSeverity));
+                    if (blink1.fadeToRGB(100 + speed * 100, colorMap.get(maxSeverity)) == -1) {
```
Comment (indigo423) [inline src/main/java/org/opennms/onmsblink/OnmsBlink.java:106]: Deal with constant -1 and 100 correctly.

File: src/main/java/org/opennms/onmsblink/OnmsBlink.java [hunk 3]
```diff
@@ -107,7 +103,9 @@ public void run() {
...
+                        blink1 = Blink1.open();
+                    }
 
                     try {
                         Thread.sleep(75 + 75 * speed);
```
Comment (indigo423) [inline src/main/java/org/opennms/onmsblink/OnmsBlink.java:111]: Deal with constant 75 correctly.

File: src/main/java/org/opennms/onmsblink/OnmsBlink.java [hunk 4]
```diff
@@ -115,7 +113,9 @@ public void run() {
                         e.printStackTrace();
                     }
 
-                    blink1.fadeToRGB(75 + speed * 75, Color.BLACK);
+                    if (blink1.fadeToRGB(75 + speed * 75, Color.BLACK) == -1) {
```
Comment (indigo423) [inline src/main/java/org/opennms/onmsblink/OnmsBlink.java:116]: Deal with constant -1 and 75 correctly.

File: src/main/java/org/opennms/onmsblink/OnmsBlink.java [hunk 5]
```diff
@@ -115,7 +113,9 @@ public void run() {
...
+                        blink1 = Blink1.open();
+                    }
 
                     try {
                         Thread.sleep(100 + 50 * speed);
```
Comment (indigo423) [inline src/main/java/org/opennms/onmsblink/OnmsBlink.java:121]: Deal with constant 50, 100 correctly.

File: src/main/java/org/opennms/onmsblink/OnmsBlink.java [hunk 6]
```diff
@@ -92,12 +85,15 @@ public boolean isConstantGlow() {
...
             while (true) {
                 if (maxSeverity.getId() <= 3 || isConstantGlow()) {
-                    blink1.fadeToRGB(0, colorMap.get(maxSeverity));
+                    System.out.println(blink1.fadeToRGB(0, colorMap.get(maxSeverity)));
+                    if (blink1.fadeToRGB(0, colorMap.get(maxSeverity)) == -1) {
+                        blink1 = Blink1.open();
+                    }
 
                     try {
                         Thread.sleep(1000);
```
Comment (indigo423) [inline src/main/java/org/opennms/onmsblink/OnmsBlink.java:99]: Deal with constant 1000 correctly.

File: src/main/java/org/opennms/onmsblink/OnmsBlink.java [hunk 7]
```diff
@@ -107,7 +103,9 @@ public void run() {
                 } else {
                     int speed = 7 - Math.max(maxSeverity.getId(), 3);
```
Comment (indigo423) [inline src/main/java/org/opennms/onmsblink/OnmsBlink.java:104]: Deal with constant 3 and 7 correctly.

**Comment by indigo423** [comment]
The while ... true loop with redundant open() seems like code smell, see: https://github.com/opennms-forge/onmsblink/blob/features/plugnplay/src/main/java/org/opennms/onmsblink/OnmsBlink.java#L91

Plug'n'Play functionality of the blink1 works. There are 9 and -1 outputs on the console when you run the tool. Is this intended?

**Comment by agalue** [comment]
Public pull requests have to be on a common language (it is my humble opinion), as this might be interpreted as you guys don't care about the opinion of non-german speakers.

**Comment by indigo423** [comment]
@agalue: Translated comments, sorry :)

**Pull request merged**